liquibase-standard/src/main/java/liquibase/database/Database.java

```diff
@@ -182,6 +196,103 @@ default String getDisplayName() {
      */
     void dropDatabaseObjects(CatalogAndSchema schema) throws LiquibaseException;
 
+    default void dropDatabaseObjects(CatalogAndSchema schemaToDrop, SnapshotControl snapshotControl) throws LiquibaseException {
+        ObjectQuotingStrategy currentStrategy = this.getObjectQuotingStrategy();
+        this.setObjectQuotingStrategy(ObjectQuotingStrategy.QUOTE_ALL_OBJECTS);
+        try {
+            DatabaseSnapshot snapshot;
+            try {
+                final Set<Class<? extends DatabaseObject>> typesToInclude = snapshotControl.getTypesToInclude();
```

Comment:
I see that you added the "null check" for snapshotControl on AbstractJdbcDatabase (creating a new one). But as this method may be called by other pieces of the code now, I think you could move the null check and the new SnapshotControl here.